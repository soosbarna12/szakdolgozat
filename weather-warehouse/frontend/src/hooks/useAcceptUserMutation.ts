import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";


export function useAcceptUserMutation() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (id: number) => {
      const response = await axios.post(`/user/accept`, { id });
      return response.data;
    },
    onSuccess: () => {
      showAlert("User accepted successfully", "success");
    },
    onError: () => {
      showAlert("Failed to accept user", "error");
    },
  });

  return mutation;
}
