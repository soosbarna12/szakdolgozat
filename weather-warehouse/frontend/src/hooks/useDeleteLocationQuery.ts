import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";

export function useDeleteLocationQuery() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (userLocationID: number) => {
      const response = await axios.delete(`/user/deleteLocation`, { params: { userLocationID } });
      return response.data;
    },
    onSuccess: () => {
      showAlert("Location deleted successfully", "success");
    },
    onError: () => {
      showAlert("Failed to delete location", "error");
    },
  });

  return mutation;
}