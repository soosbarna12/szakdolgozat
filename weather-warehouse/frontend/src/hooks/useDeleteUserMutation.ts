import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";


export function useDeleteUserMutation() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (id: number) => {
      const response = await axios.delete(`/user/delete`, { params: { id } });
      return response.data;
    },
    onSuccess: () => {
      showAlert('User deleted successfully', 'success');
    },
    onError: () => {
      showAlert('Failed to delete user', 'error');
    },
  });

  return mutation;
}
