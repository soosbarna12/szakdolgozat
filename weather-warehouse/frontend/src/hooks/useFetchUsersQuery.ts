import { useQuery } from "@tanstack/react-query";
import { User } from "../types/userData.type";
import axios from "../utils/axiosConfig";


export function useFetchUsersQuery() {
  
  const { data, error, isLoading, refetch } = useQuery<User[]>({
    queryKey: ["users"],
    queryFn: async () => {
      const response = await axios.get(`/api//user/userData`);

      return response.data as User[];
    },
  });

  return { data, error, isLoading, refetch };
}
