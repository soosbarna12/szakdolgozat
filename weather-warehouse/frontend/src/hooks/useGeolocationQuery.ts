import { useQuery } from "@tanstack/react-query";
import axios from "../utils/axiosConfig";
import { Pages } from "../types/page.type";


/*
export function useGeolocationQuery(location: string) {
    
  const { data, error, isLoading } = useQuery({
    queryKey: ["location", location],
    queryFn: async () => {
      const response = await axios.get(`/geo/location`, {
        params: { location }
      })

      if (!response?.data || response?.data?.length === 0) {
        return []; // return empty array if no data found
      }

      // search for unique locations
      const newResults = response.data.filter(
        (loc: any, index: number, self: any[]) =>
          index ===
          self.findIndex(
            (t) => t.name === loc.name && t.state === loc.state && t.country === loc.country
          )
      );
      return newResults; 
    },

    enabled: !!location,
  });

  return { data, error, isLoading };
}
*/

export function useGeolocationQuery(locationName: string, type: Pages) {
  const { data, error, isLoading } = useQuery({
    queryKey: ["locationName", locationName],
    queryFn: async () => {
      if (!locationName.trim()) {
        return []; // Return an empty array if no city name is provided
      }

      try {
        const response = await axios.get(`/geo/location`, {
          params: {
            location: locationName.trim(),
           },
        });

        if (!response?.data || response?.data?.length === 0) {
          return []; // Return an empty array if no data is found
        }

        // Filter unique city results
        const uniqueResults = response.data.filter(
          (loc: any, index: number, self: any[]) =>
            index ===
            self.findIndex(
              (t) => t.name === loc.name && t.state === loc.state && t.country === loc.country
            )
        );
        //console.log("Unique results:", uniqueResults); // Log the unique results for debugging

        return uniqueResults;
      } catch (error) {
        console.error("Error fetching geolocation data:", error);
        throw new Error("Failed to fetch geolocation data");
      }
    },
    enabled: (locationName.length !== 0 && type === Pages.Today),
    refetchOnWindowFocus: false, // Prevent refetching when the window regains focus
  });

  return { data, error, isLoading };
}