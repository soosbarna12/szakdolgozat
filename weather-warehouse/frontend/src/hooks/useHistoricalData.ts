import { useState, useEffect } from "react";
import { UseHistoricalDataProps } from "../pages/HistoricalPage/HistoricalPage.type";
import { HistoricalDataTable } from "../types/historicalDataTable.type";
import { kelvinToCelsius } from "../utils/dataConverters";


export function useHistoricalData(props: Readonly<UseHistoricalDataProps>) {
  const [tableData, setTableData] = useState<HistoricalDataTable[]>([]);
  const { data, date: selectedDate } = props;

  useEffect(() => {
    if (data?.main?.temp_max && data?.main?.temp_min) {
      const currentDate = new Date().toISOString().split("T")[0];
      
      const newRecord = {
        date: selectedDate ? selectedDate.format("YYYY-MM-DD") : currentDate,
        maxTemp: kelvinToCelsius(data.main.temp_max),
        minTemp: kelvinToCelsius(data.main.temp_min),
        humidity: data.main.humidity,
        location: [data.name, data.state, data.sys?.country].filter(Boolean).join(", "),
      };

      setTableData((prev) => {
        const isDuplicate = prev.some((record) => record.date === newRecord.date && record.location === newRecord.location);
        return isDuplicate ? prev : [...prev, newRecord];
      });
    }
  }, [data, selectedDate]);

  return { tableData };
}
