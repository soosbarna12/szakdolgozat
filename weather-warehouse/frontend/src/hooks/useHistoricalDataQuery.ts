import { useQuery } from "@tanstack/react-query";
import { UseHistoricalDataQueryProps } from "../pages/HistoricalPage/HistoricalPage.type";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";
import { useEffect } from "react";
import { convertServerHistoricalData } from "../utils/dataConverters";

export function useHistoricalDataQuery(props: Readonly<UseHistoricalDataQueryProps>) {
  const { showAlert } = useAlert();
  const { location, date } = props;
  
  const { data, error, isLoading } = useQuery({
    queryKey: ["historicalWeatherData", location, date],
    queryFn: async () => {
      const response = await axios.post(`/historical/historicalData`, {
        location,
        date
      });

      return convertServerHistoricalData(response.data)?.[0] ?? {};
    },
    enabled: !!(location && date),
  });

  useEffect(() => {
    if (error) {
      showAlert("Error fetching historical weather data", "error");
    }
  }, [error]);

  return { data, error, isLoading };
}
