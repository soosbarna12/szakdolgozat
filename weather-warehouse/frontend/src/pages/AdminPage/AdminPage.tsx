import { Box, Button, Paper, Skeleton, Table, TableBody, TableCell, TableHead, TableRow, Typography } from '@mui/material';
import React from 'react';
import { useAcceptUserMutation } from '../../hooks/useAcceptUserMutation';
import { useDeleteUserMutation } from '../../hooks/useDeleteUserMutation';
import { useFetchUsersQuery } from '../../hooks/useFetchUsersQuery';
import { ContentBox } from '../../stlyes/content.style';
import { User } from '../../types/userData.type';


export function AdminPage() {
  const { mutateAsync: acceptUserMutatue } = useAcceptUserMutation();
  const { mutateAsync: deleteUserMutatue } = useDeleteUserMutation();
  const { data: users, error, isLoading, refetch } = useFetchUsersQuery();


  // await for the acceptUserMutation to complete before refetching
  // load userdata before accepting or deleting a user (usernotfound error)
  async function handleAcceptUser(id: number) {
    acceptUserMutatue(id)?.then(() => { refetch() });
  }

  async function handleDeleteUser(id: number) {
    deleteUserMutatue(id)?.then(() => { refetch() });
  }


  function renderShimmerTableContent() {
    for (let i = 0; i < 5; i++) {
      return (
        <TableRow key={i} data-testid="shimmerTableRow">
          <TableCell>
            <Skeleton variant="text" width={50} />
          </TableCell>
          <TableCell>
            <Skeleton variant="text" width={100} />
          </TableCell>
          <TableCell>
            <Skeleton variant="text" width={80} />
          </TableCell>
          <TableCell align="right">
            <Skeleton variant="rectangular" width={100} height={30} />
          </TableCell>
        </TableRow>
      );
    }
  }

  function renderShimmer() {
    if (isLoading) {
      return (
        <ContentBox>
          <Box p={3}>
            <Typography variant="h4" gutterBottom data-testid="userManagementTitle">
              User Management
            </Typography>
            <Paper>
              <Table>
                <TableHead>
                  <TableRow data-testid="shimmerTableRowHeader">
                    <TableCell>User ID</TableCell>
                    <TableCell>Username</TableCell>
                    <TableCell>Status</TableCell>
                    <TableCell align="right">Actions</TableCell>
                  </TableRow>
                </TableHead>
                <TableBody>
                  {renderShimmerTableContent()}
                </TableBody>
              </Table>
            </Paper>
          </Box>
        </ContentBox>
      );
    }
  }

  function renderError() {
    if (error) {
      return (
        <p>Error fetching data.</p>
      );
    }
  }

  function renderAcceptButton(user: User) {
    if (user.status === 'pending') {
      return (
        <Button
          variant="outlined"
          color="primary"
          sx={{ mr: 1 }}
          onClick={() => handleAcceptUser(user.userId)}
        >
          Accept
        </Button>
      );
    }
  }

  function renderDeleteButton(user: User) {
    if (!user.isAdmin) {
      return (
        <Button
          variant="outlined"
          color="error"
          onClick={() => handleDeleteUser(user.userId)}
        >
          Delete
        </Button>
      );
    }
  }

  function renderTableContent() {
    if (users && users.length > 0) {
      return users.map((user) => (
        <TableRow key={user.userId}>
          <TableCell>{user.userId}</TableCell>
          <TableCell>{user.username}</TableCell>
          <TableCell>{user.status}</TableCell>
          <TableCell align="right">
            {renderAcceptButton(user)}
            {renderDeleteButton(user)}
          </TableCell>
        </TableRow>
      ));
    }
  }

  function renderTable() {
    return (
      <ContentBox>
        <Box p={3}>
          <Typography variant="h4" gutterBottom>
            User Management
          </Typography>
          <Paper>
            <Table>
              <TableHead>
                <TableRow>
                  <TableCell>User ID</TableCell>
                  <TableCell>Username</TableCell>
                  <TableCell>Status</TableCell>
                  <TableCell align="right">Actions</TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                {renderTableContent()}
              </TableBody>
            </Table>
          </Paper>
        </Box>
      </ContentBox>
    );
  }

  return (
    <>
      {renderShimmer()}
      {renderError()}
      {renderTable()}
    </>
  )
}
