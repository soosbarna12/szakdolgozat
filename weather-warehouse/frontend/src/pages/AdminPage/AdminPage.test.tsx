import '@testing-library/jest-dom';
import { fireEvent, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { useAcceptUserMutation } from '../../hooks/useAcceptUserMutation';
import { useDeleteUserMutation } from '../../hooks/useDeleteUserMutation';
import { useFetchUsersQuery } from '../../hooks/useFetchUsersQuery';
import { renderWithQueryClient } from '../../utils/test/renderWithQueryClient';
import { AdminPage } from './AdminPage';

jest.mock('../../hooks/useAcceptUserMutation', () => ({
  useAcceptUserMutation: jest.fn(),
}));

jest.mock('../../hooks/useDeleteUserMutation', () => ({
  useDeleteUserMutation: jest.fn(),
}));

jest.mock('../../hooks/useFetchUsersQuery', () => ({
  useFetchUsersQuery: jest.fn(),
}));

// Mock react-leaflet components
jest.mock('react-leaflet', () => ({
  MapContainer: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="dataMapContainer">{children}</div>
  ),
  TileLayer: () => <div data-testid="mockTileLayer">Mock TileLayer</div>,
}));

describe('pages/AdminPage', () => {
  const mockAcceptUserMutation = jest.fn().mockResolvedValueOnce({})
  const mockDeleteUserMutation = jest.fn().mockResolvedValueOnce({})
  const mockRefetch = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    (useAcceptUserMutation as jest.Mock).mockReturnValue({
      mutateAsync: mockAcceptUserMutation,
    });
    (useDeleteUserMutation as jest.Mock).mockReturnValue({
      mutateAsync: mockDeleteUserMutation,
    });
    (useFetchUsersQuery as jest.Mock).mockReturnValue({
      data: [
        { userId: 1, username: 'user1', status: 'pending', isAdmin: false },
        { userId: 2, username: 'user2', status: 'active', isAdmin: false },
      ],
      error: null,
      isLoading: false,
      refetch: mockRefetch,
    });
  });

  it('matches AdminPage component snapshot', () => {
    const result = renderWithQueryClient(<AdminPage />);
    expect(result.container).toMatchSnapshot();
  });

  it('renders render shimmer loading state when loading', async () => {
    (useFetchUsersQuery as jest.Mock).mockReturnValue({
      data: null,
      error: null,
      isLoading: true,
      refetch: mockRefetch,
    });

    const result = renderWithQueryClient(<AdminPage />);

    expect(result.container).toMatchSnapshot();
  });

  it('renders error message when there is an error', () => {
    (useFetchUsersQuery as jest.Mock).mockReturnValue({
      data: null,
      error: true,
      isLoading: false,
      refetch: mockRefetch,
    });

    renderWithQueryClient(<AdminPage />);

    expect(screen.getByText('Error fetching data.')).toBeInTheDocument();
  });

  it('calls accept user mutation and refetches data', async () => {
    renderWithQueryClient(<AdminPage />);

    const acceptButton = screen.getAllByText('Accept');
    fireEvent.click(acceptButton[0]);

    await waitFor(() => {
      expect(mockAcceptUserMutation).toHaveBeenCalledWith(1);
      expect(mockRefetch).toHaveBeenCalled();
    });
  });

  it('calls delete user mutation and refetches data', async () => {
    renderWithQueryClient(<AdminPage />);

    const deleteButton = screen.getAllByText('Delete');
    fireEvent.click(deleteButton[0]);

    await waitFor(() => {
      expect(mockDeleteUserMutation).toHaveBeenCalledWith(1);
      expect(mockRefetch).toHaveBeenCalled();
    });
  });

  it('does not render delete button for admin users', () => {
    (useFetchUsersQuery as jest.Mock).mockReturnValue({
      data: [
        { userId: 1, username: 'admin', status: 'active', isAdmin: true },
      ],
      error: null,
      isLoading: false,
      refetch: mockRefetch,
    });

    renderWithQueryClient(<AdminPage />);

    expect(screen.queryByText('Delete')).toBeNull();
  });
});