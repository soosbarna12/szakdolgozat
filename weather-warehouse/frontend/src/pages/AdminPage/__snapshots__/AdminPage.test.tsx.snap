// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/AdminPage matches AdminPage component snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-uasic2"
  >
    <div
      class="MuiBox-root css-19kzrtu"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1it4a80-MuiTypography-root"
      >
        User Management
      </h4>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1gtchvp-MuiPaper-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <table
          class="MuiTable-root css-1xwxv7r-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                User ID
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                Username
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                Status
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-to33az-MuiTableCell-root"
                scope="col"
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                1
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                user1
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                pending
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-3ssuu9-MuiTableCell-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-ox4pna-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Accept
                </button>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorError MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorError css-1hz6i2n-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                2
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                user2
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                active
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-3ssuu9-MuiTableCell-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorError MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorError css-1hz6i2n-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Delete
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`pages/AdminPage renders render shimmer loading state when loading 1`] = `
<div>
  <div
    class="MuiBox-root css-uasic2"
  >
    <div
      class="MuiBox-root css-19kzrtu"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1it4a80-MuiTypography-root"
        data-testid="userManagementTitle"
      >
        User Management
      </h4>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1gtchvp-MuiPaper-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <table
          class="MuiTable-root css-1xwxv7r-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
              data-testid="shimmerTableRowHeader"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                User ID
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                Username
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                Status
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-to33az-MuiTableCell-root"
                scope="col"
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
              data-testid="shimmerTableRow"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
                  style="width: 50px;"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
                  style="width: 100px;"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
                  style="width: 80px;"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-3ssuu9-MuiTableCell-root"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
                  style="width: 100px; height: 30px;"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-uasic2"
  >
    <div
      class="MuiBox-root css-19kzrtu"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1it4a80-MuiTypography-root"
      >
        User Management
      </h4>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1gtchvp-MuiPaper-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <table
          class="MuiTable-root css-1xwxv7r-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                User ID
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                Username
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                Status
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-to33az-MuiTableCell-root"
                scope="col"
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </div>
</div>
`;
